src/app/utils/url-generator.js

```diff
@@ -30,6 +30,18 @@ export const profileSandboxesUrl = (username: string, page?: number) =>
 export const profileLikesUrl = (username: string, page?: number) =>
   `${profileUrl(username)}/likes${page ? `/${page}` : ''}`;
 
+export const githubRepoUrl = ({
+  repo,
+  branch,
+  username,
+  path,
+}: {
+  repo: string,
+  branch: string,
+  username: string,
+  path: string,
+}) => `https://github.com/${username}/${repo}/tree/${branch}/${path}`;
+
 export const optionsToParameterizedUrl = (options: Object) => {
```

Comment:
You need to encode your keys and values. You should also sort so that your cache is more predictable.
```js
export const optionsToParameterizedUrl = (options: Object) => {
  const keyValues = Object.keys(options)
     .sort()
     .map(key => `${encodeURIComponent(key)}=${encodeURIComponent(options[key])}`)
     .join('&');
  return keyValues ? `?${keyValues}` : '';
 };
```